### PR TITLE
Extract all command

### LINKED ISSUE #1: Need to extract all attachments
Feature request:

Allow passing * to allow for all attachments not just int value

### PR DESCRIPTION
**Pull request by debragail**
added --extract-all command

### CODE REVIEW

File: eml_analyzer/cli_script.py [hunk 1]
```diff
@@ -181,8 +180,27 @@ def extract_attachment(parsed_eml: Message, attachment_number: int, output_path:
...
+    for child in parsed_eml.walk():
+      if child.get_filename() is None:
+        continue
+      print(child.get_filename())
+      path = output_path
+      attachment = child      
+      if path is None:
+          path = attachment.get_filename()
+      elif os.path.isdir(path):
+          path = os.path.join(path, attachment.get_filename())
```
Comment (wahlflo) [inline eml_analyzer/cli_script.py:197]: check case where **path** is a path to file. This case is currently missing

File: eml_analyzer/cli_script.py [hunk 2]
```diff
@@ -181,8 +180,27 @@ def extract_attachment(parsed_eml: Message, attachment_number: int, output_path:
...
-
+    
+    
+def extract_all_attachments(parsed_eml: Message, output_path: str or None):
+    print_headline_banner('Attachment Extracting')
+    attachment = None
+    for child in parsed_eml.walk():
+      if child.get_filename() is None:
+        continue
+      print(child.get_filename())
```
Comment (wahlflo) [inline eml_analyzer/cli_script.py:191]: Maybe combine this print out with the print out in line 201: **info('Attachment extracted to {}'.format(path))**

Recommendation:
`info('Attachment "{}" extracted to {}'.format(attachment.get_filename(), path))`

File: setup.py [hunk 3]
```diff
@@ -6,13 +6,13 @@
 
 setuptools.setup(
     name="eml-analyzer",
-    version="1.0.1",
-    author="Florian Wahl",
-    author_email="florian.wahl.developer@gmail.com",
+    version="2.0.0",
+    author="Debra White",
```
Comment (wahlflo) [inline setup.py:10]:
```suggestion
    author="Florian Wahl",
```

File: setup.py [hunk 4]
```diff
@@ -6,13 +6,13 @@
...
+    author_email="git@27c1ub.com",
```
Comment (wahlflo) [inline setup.py:11]:
```suggestion
    author_email="florian.wahl.developer@gmail.com",
```

File: setup.py [hunk 5]
```diff
@@ -6,13 +6,13 @@
...
     description="A cli script to analyze an E-Mail in the eml format for viewing the header, extracting attachments etc.",
     long_description=long_description,
     long_description_content_type="text/markdown",
-    url="https://github.com/wahlflo/eml_analyzer",
+    url="https://github.com/debragail/eml_analyzer",
```
Comment (wahlflo) [inline setup.py:15]:
```suggestion
    url="url="https://github.com/wahlflo/eml_analyzer",
```

**Review by wahlflo — CHANGES_REQUESTED** [review]
Hi, 
thank you for your contribution.
Please delete your uploaded PDFs and the email. 
I requested also some changes for the python code itself.
And the setup.py should also stay the same when merging it - exept the version ;) .

**Comment by wahlflo** [comment]
added the --extract-all command in the commit 82977e719005bd50bd5f8773861e3e9ef162fc53

**Comment by debragail** [comment]
thanks

**Pull request closed without merging**